### PR TITLE
AdvectionDiffusionProblem: restructure

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/meltpooldg/advection_diffusion/advection_diffusion_problem.hpp [hunk 1]
```diff
@@ -164,18 +158,36 @@ namespace MeltPoolDG
...
         scratch_data->build();
 
-        // TODO: better place
-        if (advec_diff_operation)
-          {
-            scratch_data->initialize_dof_vector(advec_diff_operation->get_advected_field());
-            scratch_data->initialize_dof_vector(advec_diff_operation->get_advected_field_old());
-          }
+        if (advec_diff_operation) // TODO: better place
+          advec_diff_operation->reinit();
```
Comment (peterrum) [inline include/meltpooldg/advection_diffusion/advection_diffusion_problem.hpp:162]: TODO

File: include/meltpooldg/advection_diffusion/advection_diffusion_problem.hpp [hunk 2]
```diff
@@ -278,13 +283,11 @@ namespace MeltPoolDG
...
-                *scratch_data,
-                dof_zero_bc_idx,
-                quad_idx,
-                dof_idx_velocity,
-                initial_solution,
-                advection_velocity,
-                base_in);
+                *scratch_data, dof_zero_bc_idx, quad_idx, dof_idx_velocity, base_in);
+
+            advec_diff_operation->reinit();
```
Comment (peterrum) [inline include/meltpooldg/advection_diffusion/advection_diffusion_problem.hpp:288]: TODO: should happen by the above function! For that the two implementations have to be unified.

**Review by peterrum — COMMENTED** [review]
I would like to point out two TODOs.

**Comment by mschreter** [comment]
Looks good to me!

**Pull request merged**